### PR TITLE
Added api docs gen (copy from TFQ).

### PR DESCRIPTION
**Pull request by MichaelBroughton**
First pass on docs build for API (Copies from TFQ). Goes along with #3085.
@yashk2810
@lamberta
I included the built copies of these docs (took all of 10s to build WOW!) for ease of testing, but we should probably remove them before merging.

### CODE REVIEW

File: docs/build_api_docs.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 The Cirq Authors. All Rights Reserved.
```
Comment (lamberta) [inline docs/build_api_docs.py:1]: Can please you move this file into `dev_tools/docs/` ?

Comment (MichaelBroughton) [reply docs/build_api_docs.py]: Done.

File: dev_tools/docs/build_api_docs.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        root_title="Cirq",
+        py_modules=[("cirq", cirq)],
+        base_dir=os.path.dirname(cirq.__file__),
+        code_url_prefix=FLAGS.code_url_prefix,
+        search_hints=FLAGS.search_hints,
+        site_path=FLAGS.site_path,
+        callbacks=[public_api.local_definitions_filter],
+        private_map={
+            "cirq.google.engine.client.quantum.QuantumEngineServiceClient" : ["enums"],
+            "cirq.google.engine.client.quantum_v1alpha1.QuantumEngineServiceClient":["enums"]
```
Comment (dstrain115) [inline dev_tools/docs/build_api_docs.py:58]: I think this line may be too long.  Can you run check/format in order to fix the tests?

File: dev_tools/docs/build_api_docs.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tool to generate external api_docs for Cirq (Shameless copy from TFQ)."""
+
+from __future__ import absolute_import
```
Comment (dabacon) [inline dev_tools/docs/build_api_docs.py:17]: are these future imports necessay since we don't support python < 3.6?

Comment (yashk2810) [reply dev_tools/docs/build_api_docs.py:17]: +1

File: dev_tools/docs/build_api_docs.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    doc_generator = generate_lib.DocGenerator(
+        root_title="Cirq",
+        py_modules=[("cirq", cirq)],
+        base_dir=os.path.dirname(cirq.__file__),
+        code_url_prefix=FLAGS.code_url_prefix,
+        search_hints=FLAGS.search_hints,
+        site_path=FLAGS.site_path,
+        callbacks=[public_api.local_definitions_filter],
+        private_map={
+            "cirq.google.engine.client.quantum.QuantumEngineServiceClient" : ["enums"],
```
[the remark this answers is not in the document]
Comment (yashk2810) [reply dev_tools/docs/build_api_docs.py:57]: private_map is used to NOT document certain parts of the API. 

https://source.corp.google.com/piper///depot/google3/third_party/py/tensorflow_docs/api_generator/generate_lib.py;l=746

**Pull request merged**